packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -502,6 +509,34 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
     }
   }
 
+  static _findItemIndexWithKey(
+    props: Props,
+    key: string,
+    hint: ?number,
+  ): ?number {
+    if (hint != null) {
+      const curKey = VirtualizedList._getItemKey(props, hint);
+      if (curKey === key) {
+        return hint;
+      }
+    }
+    for (let ii = 0; ii < props.getItemCount(props.data); ii++) {
+      const curKey = VirtualizedList._getItemKey(props, ii);
+      if (curKey === key) {
+        return ii;
+      }
+    }
+    return null;
+  }
+
+  static _getItemKey(props: Props, index: number): ?string {
+    const item = props.data ? props.getItem(props.data, index) : null;
```

Comment:
This would cause `renders null list` test to fail since we use it to set initial state. I could move the null check to problematic call site too.

Not sure exactly what you mean by `There is code out there with nullish data but getItem hardcoding values.`